### ISSUE TITLE
On MacOS, the py_image_base tar files that I get has the wrong permissions 

### ISSUE THREAD

**Issue by kovkev** [bazelbuild/rules_docker#1265]
I use bazel-buildfarm to build the images and send them to my local machine.

I made sure `/tmp/some/` has recursively `chmod -R agu+rwx /tmp/some/` all permissions.

However, when bazel tries to get the image from bazel-buildfarm, I think it does `touch 002.tar.gz.nogz` for each file and does not set the right permissions. 

```
Users-MacBook-Pro:__main__ kevin$ ls -lah /tmp/some/execroot/__main__/bazel-out/darwin-fastbuild/bin/external/py_image_base/
total 0
drwxrwxrwx  3 kevin  wheel    96B Nov  8 08:50 .
drwxrwxrwx  3 kevin  wheel    96B Nov  8 08:50 ..
drwxrwxrwx  3 kevin  wheel    96B Nov  8 08:56 image
Users-MacBook-Pro:__main__ kevin$ ls -lah /tmp/some/execroot/__main__/bazel-out/darwin-fastbuild/bin/external/py_image_base/image
total 0
drwxrwxrwx  3 kevin  wheel    96B Nov  8 08:56 .
drwxrwxrwx  3 kevin  wheel    96B Nov  8 08:50 ..
-r-xr-xr-x  1 kevin  wheel     0B Nov  8 08:56 002.tar.gz.nogz
```

I dug a little too much in that direction the last couple of days so I'll take a break on this issue.

**Comment by nlopezgi** [comment]
Hi @kovkev ,
we dont really have a way to repro buildfarm issues, can you try to repro this with a local build so we can help? Otherwise I'm not sure we can provide much assistance.

**Comment by kovkev** [comment]
Hi @nlopezgi , without using buildfarm, the issue does not happen.

Here is what I get with buildfarm:
```
Users-MacBook-Pro:tgkj kevin$ bazel build --verbose_failures --verbose_explanations --explain=/tmp/explain //analysis/thinking:jupyter   --extra_execution_platforms=//toolchains:my_platform,@local_config_platform//:host,@io_bazel_rules_docker//platforms:local_container_platform --logging=6 --experimental_repository_resolved_file=/tmp/a --nolegacy_important_outputs --experimental_remote_grpc_log=/tmp/grpc_log --execution_log_json_file=/tmp/spawns -s
INFO: Writing tracer profile to '/private/tmp/some/command.profile.gz'
INFO: Invocation ID: c8a948a7-0c46-4e0b-be99-53c34686679d
INFO: Analyzed target //analysis/thinking:jupyter (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
INFO: Writing explanation of rebuilds to '/tmp/explain'
SUBCOMMAND: # @py_image_base//image:image [action 'GUNZIP external/py_image_base/image/002.tar.gz.nogz', configuration: 79c79b07561f62ef3c67caefaa03e7af]
(cd /private/tmp/some/execroot/__main__ && \
  exec env - \
    PATH=/Users/kevin/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/kevin/kg_bin/:/Users/kevin/bin \
  /bin/bash -c 'bazel-out/host/bin/external/gzip/gzip -d < external/py_image_base/image/002.tar.gz > bazel-out/darwin-fastbuild/bin/external/py_image_base/image/002.tar.gz.nogz')
SUBCOMMAND: # @py_image_base//image:image [action 'GUNZIP external/py_image_base/image/001.tar.gz.nogz', configuration: 79c79b07561f62ef3c67caefaa03e7af]
(cd /private/tmp/some/execroot/__main__ && \
  exec env - \
    PATH=/Users/kevin/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/kevin/kg_bin/:/Users/kevin/bin \
  /bin/bash -c 'bazel-out/host/bin/external/gzip/gzip -d < external/py_image_base/image/001.tar.gz > bazel-out/darwin-# build --spawn_strategy=remote --genrule_strategy=remote --strategy=Javac=remote --strategy=Closure=remote --remote_efastbuild/bin/external/py_image_base/image/001.tar.gz.nogz')
SUBCOMMAND: # @py_image_base//image:image [action 'GUNZIP external/py_image_base/image/000.tar.gz.nogz', configuration: 79c79b07561f62ef3c67caefaa03e7af]
(cd /private/tmp/some/execroot/__main__ && \
  exec env - \
    PATH=/Users/kevin/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/kevin/kg_bin/:/Users/kevin/bin \
  /bin/bash -c 'bazel-out/host/bin/external/gzip/gzip -d < external/py_image_base/image/000.tar.gz > bazel-out/darwin-fastbuild/bin/external/py_image_base/image/000.tar.gz.nogz')
SUBCOMMAND: # @py_image_base//image:image [action 'GUNZIP external/py_image_base/image/003.tar.gz.nogz', configuration: 79c79b07561f62ef3c67caefaa03e7af]
(cd /private/tmp/some/execroot/__main__ && \
  exec env - \
    PATH=/Users/kevin/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/kevin/kg_bin/:/Users/kevin/bin \
  /bin/bash -c 'bazel-out/host/bin/external/gzip/gzip -d < external/py_image_base/image/003.tar.gz > bazel-out/darwin-fastbuild/bin/external/py_image_base/image/003.tar.gz.nogz')
ERROR: /private/tmp/some/external/py_image_base/image/BUILD:4:1: error executing shell command: '/bin/bash -c bazel-out/host/bin/external/gzip/gzip -d < external/py_image_base/image/002.tar.gz > bazel-out/darwin-fastbuild/bin/external/py_image_base/image/002.tar.gz.nogz' failed (Exit 126)
/bin/bash: bazel-out/host/bin/external/gzip/gzip: cannot execute binary file: Exec format error
Target //analysis/thinking:jupyter failed to build
INFO: Elapsed time: 1.132s, Critical Path: 0.53s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

It has to do with, I think, `rules_docker` doing `touch 002.tar.gz.nogz`, or some other tool that `rules_docker` is using is doing `touch 002.tar.gz.nogz` . I don't know how to figure out what tool is doing so.

When the touch happens, it should be able to set the writing permissions to writeable by bazel at least...

**Comment by darl** [comment]
I have the same issue with buildbarn.

I think it attempts to run `gzip` binary on remote machine (linux) uploaded from host machine (macos).
https://github.com/bazelbuild/rules_docker/blob/738ea4f4b7baf00383b553149b692948310f9e7a/repositories/repositories.bzl#L244-L247

Looking for workarounds...

**Comment by nlopezgi** [comment]
thanks for noting this.
I just noticed this might be related to https://github.com/bazelbuild/rules_docker/issues/764. In both cases the behavior of these actions is not the expected one. My running hypothesis is that `local_tool` does not work properly with remote execution. The solution is likely going to require folding in gzip into the docker toolchain, as is the case for `xz` (see https://github.com/bazelbuild/rules_docker/blob/master/toolchains/docker/toolchain.bzl#L27). 

Note this is not something we are currently planning on working on, but I'd be happy to work with someone that wants to volunteer a PR for this.

**Comment by nlopezgi** [comment]
I created https://github.com/bazelbuild/rules_docker/pull/1272 that should fix this for Starlark rules (not for repo rules). Please give it a try and see if that addresses your issue.

**Comment by nlopezgi** [comment]
Closing as this should be fixed by #1272

**Comment by gh-atuzhikov** [comment]
Hi, I am also trying to setup buildfarm to speedup building images, may I ask you, @nlopezgi, @kovkev and @darl, how does one configure the workers? what I see currently, is that the jobs get submitted, but then the images never start building. I am ssuming, that the platform values are not set correctly below? Thank you in advance!
`port: 8981
public_name: "localhost:8981"
omit_from_cas: false
digest_function: SHA256
root: "/tmp/worker"
inline_content_limit: 1048567 # 1024 * 1024

operation_poll_period: {

  seconds: 1
  nanos: 0

}

dequeue_match_settings: {

  platform: {
  }

}

cas: {

  filesystem: {
  
    path: "cache"
    max_size_bytes: 2147483648 # 2 * 1024 * 1024 * 1024
    max_entry_size_bytes: 2147483648 # 2 * 1024 * 1024 * 1024
    file_directories_index_in_memory: false

  }

  skip_load: false
}

execute_stage_width: 1

input_fetch_stage_width: 1

link_input_directories: true

default_action_timeout: {
  seconds: 36000
  nanos: 0
}

maximum_action_timeout: {
  seconds: 3600
  nanos: 0
}

redis_shard_backplane_config: {

  redis_uri: "redis://localhost:6379"
  jedis_pool_max_total: 4000
  workers_hash_name: "Workers"
  worker_channel: "WorkerChannel"
  action_cache_prefix: "ActionCache"
  action_cache_expire: 2419200 # 4 weeks
  action_blacklist_prefix: "ActionBlacklist"
  action_blacklist_expire: 3600 # 1 hour
  invocation_blacklist_prefix: "InvocationBlacklist"
  operation_prefix: "Operation"
  operation_expire: 604800 # 1 week
  queued_operations_list_name: "{Execution}:QueuedOperations"
  dispatching_prefix: "Dispatching"
  dispatching_timeout_millis: 10000
  dispatched_operations_hash_name: "DispatchedOperations"
  operation_channel_prefix: "OperationChannel"
  cas_prefix: "ContentAddressableStorage"
  cas_expire: 604800 # 1 week
  subscribe_to_backplane: true

  provisioned_queues: {
    queues: {
      name: "cpu"

      platform: {
        properties: {
          name: "min-cores"
          value: "*"
        }
        properties: {
          name: "max-cores"
          value: "*"
        }
      }
    }
  }
}

exec_owner: ""
`
